consensus/polybft/fsm.go

```diff
@@ -0,0 +1,558 @@
+package polybft
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/pbft-consensus"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+	hcf "github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+)
+
+var _ pbft.Backend = &fsm{}
+
+type blockBuilder interface {
+	Reset()
+	// CommitTransaction(tx *types.Transaction) error
+	// Fill(ctx context.Context) error
+	Build(func(h *types.Header)) (*StateBlock, error)
+	GetState() *state.Transition
+}
+
+const (
+	stateTransactionsGasLimit = 1000000 // some arbitrary default gas limit for state transactions
+	maxBundlesPerSprint       = 50
+)
+
+type fsm struct {
```

Comment:
@ferranbt @JekaMas @vcastellm 

We should create a task for breaking up this struct in the future, it's insanely beefed up and has a huge design smell

cc @Kourin1996